src/swc/dirWorker.ts

```diff
@@ -13,12 +13,13 @@ export default async function handleCompile(opts: {
   sync: boolean;
   cliOptions: CliOptions;
   swcOptions: Options;
+  outFileExtension: string;
 }) {
   const dest = getDest(
     opts.filename,
     opts.outDir,
     opts.cliOptions.stripLeadingPaths,
-    ".js"
+    `.${opts.outFileExtension}`
```

Comment:
I checked how babel cli works. It takes extension already starts with a dot. But according to this PR https://github.com/swc-project/swc/pull/4343 native swc cli already takes it without a dot.